### PR TITLE
[App proxy and Unauthenticated storefront]: Code fixes

### PR DESCRIPTION
**Pull request by mcvinci**
## This PR: 
- Fixes an incorrectly referenced constant in the [App Proxy reference documentation](https://shopify.dev/docs/api/shopify-app-remix/v1/authenticate/public/app-proxy#storefront-example-group)
- Fixes some code that references `authenticated.storefront` versus `authenticate.public.appProxy(request)`  in the [Unauthenticated storefront documentation](https://shopify.dev/docs/api/shopify-app-remix/v1/unauthenticated/unauthenticated-storefront#example-graphql)
- Fixes https://github.com/Shopify/shopify-dev/issues/39079
- Fixes https://github.com/Shopify/shopify-dev/issues/38934

### CODE REVIEW

File: packages/shopify-app-remix/src/server/clients/storefront/types.ts [hunk 1]
```diff
@@ -13,10 +13,10 @@ export interface StorefrontContext {
    * <description>Use `storefront.graphql` to make query / mutation requests.</description>
    * ```ts
    * import { ActionFunctionArgs } from "@remix-run/node";
-   * import { authenticate } from "../shopify.server";
+   * import { unauthenticated } from "../shopify.server";
```
Comment (paulomarg) [inline packages/shopify-app-remix/src/server/clients/storefront/types.ts:16]: I wonder if we should use `appProxy` as our "official" example here - we want to encourage folks to rely on the authenticated methods as much as we can.

**Review by paulomarg — APPROVED** [review]
LGTM, just had one question.

File: packages/shopify-app-remix/src/server/clients/storefront/types.ts (hunk 1 above)
Comment (mcvinci) [reply packages/shopify-app-remix/src/server/clients/storefront/types.ts]: Thanks, @paulomarg! I just added the `appProxy` example. Would you mind taking another look to confirm all looks correct?

**Review by paulomarg — APPROVED** [review]

**Pull request merged**